File: youlexuan-manager-web/src/main/webapp/js/service/typeTemplateService.js
app.service('typeTemplateService',function ($http) {

    this.findAll = function () {
        return $http.get('../typeTemplate/findAll.do');
    }

    this.findPage = function (pageNum,pageSize,entity) {
        return $http.post('../typeTemplate/findPage.do?pageNum='+pageNum+'&pageSize='+pageSize,entity);
    }

    this.selectList = function () {
        return $http.get('../typeTemplate/selectList.do');
    }

    this.add = function (entity) {
        return $http.post('../typeTemplate/add.do',entity);
    }

    this.findOne = function (id) {
        return $http.get('../typeTemplate/findOne.do?id='+id);
    }

    this.update = function (entity) {
        return $http.post('../typeTemplate/update.do',entity);
    }

    this.delete = function (ids) {
        return $http.get('../typeTemplate/delete.do?ids='+ids);
    }
})

File: youlexuan-sellergoods-interface/src/main/java/com/offcn/sellergoods/service/TypeTemplateService.java
package com.offcn.sellergoods.service;

import com.offcn.entity.PageResult;
import com.offcn.pojo.TbTypeTemplate;

import java.util.List;
import java.util.Map;

public interface TypeTemplateService {

    /**
     * 返回指定模板id的规格列表，包含规格选项
     * @param id
     * @return
     */
    public List<Map> findSpecList(Long id);
    /**
     * 查询所有
     * @return
     */
    public List<TbTypeTemplate> findAll();

    /**
     * 带条件带分页查询
     * @param pageNum
     * @param pageSize
     * @param tbTypeTemplate
     * @return
     */
    public PageResult findPage(int pageNum, int pageSize, TbTypeTemplate tbTypeTemplate);

    /**
     * 关联品牌和关联规格下拉框
     * @return
     */
    public Map selectOptionList();

    /**
     * 新建模板
     * @param tbTypeTemplate
     */
    public void add(TbTypeTemplate tbTypeTemplate);

    /**
     * 根据id获取实体
     * @param id
     * @return
     */
    public TbTypeTemplate findOne(Long id);

    /**
     * 更新模板
     * @param tbTypeTemplate
     */
    public void update(TbTypeTemplate tbTypeTemplate);

    /**
     * 批量删除
     * @param ids
     */
    public void delete(Long[] ids);

}
